Add KCSA Certification to Training page

Community-Programmer opened the pull request:
Added KCSA Certification to Training page

- Added KCSA Logo at Section page
- Added KCSA Certification information

![img](https://github.com/kubernetes/website/assets/76515568/437d5148-d034-4ab6-82c1-f96d638732e7)
![img2](https://github.com/kubernetes/website/assets/76515568/5044e306-5445-4997-8a56-a90521df1559)

aj11anuj reviewed (commented):
LGTM, 👍

Gauravpadam commented on content/en/training/_index.html:
```diff
@@ -14,6 +14,9 @@
         <h2>Build your cloud native career</h2>
         <p>Kubernetes is at the core of the cloud native movement. Training and certifications from the Linux Foundation and our training partners lets you invest in your career, learn Kubernetes, and make your cloud native projects successful.</p>
       </div>
+      <div class="logo-certification cta-image" id="logo-kcsa">
+        <img src="/images/training/kubernetes-cksa-white.svg" />
+      </div>
```
I'd move this div towards the end of logo divs

windsonsea commented:
The [preview](https://deploy-preview-43318--kubernetes-io-main-staging.netlify.app/training/) looks good, but I'm unsure which sequence would be better. Should we place two "security" icons adjacent to each other?

Community-Programmer commented:
yes I think they should be adjacent to each other.

This is how its look at end
![image](https://github.com/kubernetes/website/assets/76515568/5231f6df-3749-4fed-8584-6cd5c66d300a)

Also removed line breaks in it, now they are not aligned
![image](https://github.com/kubernetes/website/assets/76515568/f9a8f9c9-77f8-4b39-ad55-03f0b8cc7385)

aj11anuj commented:
> The [preview](https://deploy-preview-43318--kubernetes-io-main-staging.netlify.app/training/) looks good, but I'm unsure which sequence would be better.
> 
> * Would it be better to have two "security" icons placed next to each other?
> * Alternatively, placing this new icon in the 2nd position might be more suitable.

+1 to 2nd point, as this certification is a beginner level certification.. it's more suitable at 2nd place (just after KCNA). For icons as well as tiles order from left to right will be more suitable as (KCNA, KCSA, CKAD, CKA and then CKS).

Gauravpadam commented:
- I'm good with either position for the icon. Since both reviewers prefer 2nd position, Let's go with that
- The spacing can be adjusted with some css changes, That can come as a different PR

sftim commented:
Please don't mention issue references in PR titles

/retitle Add KCSA Certification to Training page

sftim commented:
We should try to make the page render better for narrow viewports - see this screenshot for why
![screenshot](https://github.com/kubernetes/website/assets/22591623/3f1bff08-1356-4ea5-b5a4-a5b00f8f7529)

@Community-Programmer do you want to add a commit to do that? If not, we can look to merge this and then file a new issue about making the page responsive.

Community-Programmer commented:
> We should try to make the page render better for narrow viewports - see this screenshot for why ![screenshot](https://user-images.githubusercontent.com/22591623/272871377-3f1bff08-1356-4ea5-b5a4-a5b00f8f7529.png)
> 
> @Community-Programmer do you want to add a commit to do that? If not, we can look to merge this and then file a new issue about making the page responsive.

Yes, I will fix that too. I will add a commit for the previous one and this one together.

sftim commented on content/en/training/_index.html:
```diff
@@ -115,11 +119,23 @@ <h5>
       </div>
+      <div class="col-nav">
+        <h5>
+          <b>Kubernetes and Cloud Native Security Associate (KCSA)</b>
```
nit: maybe put this _before_ the CKS?

Community-Programmer commented:
**Changes Made:**

- Changed the Position of logo of KCSA to 2nd Position
- Added CSS   to make the page render better for narrow viewports
- Removed Extra Spaces and <Br> tags

Images attached:

![image](https://github.com/kubernetes/website/assets/76515568/86ae2520-555b-42b1-961f-1328d16def5d)
![image](https://github.com/kubernetes/website/assets/76515568/cfb83c75-f60b-416e-a3a6-3474a069f65f)
![image](https://github.com/kubernetes/website/assets/76515568/92207c32-1582-4218-9698-3222c83e90e8)

sftim commented:
This looks OK to merge. Please squash it to 2 commits:
- the CSS change
- adding the KCSA

---

Even better (**optionally**):

change from:
![before](https://github.com/kubernetes/website/assets/22591623/f2383dd2-9845-4d41-a51b-833f3476f564)

to more like:
![after](https://github.com/kubernetes/website/assets/22591623/b9ab4686-b9d7-4a87-a322-d5579fd68510)

Gauravpadam commented on static/css/training.css:
```diff
@@ -151,7 +170,8 @@ body.cid-training #get-certified .col-nav a.button {
 @media only screen and (max-width: 480px) {
   body.cid-training .col { margin: 1% 0 1% 0%;}
-  body.cid-training .col-container { flex-direction: column; }
+  body.cid-training .col-container { display: flex; flex-direction: column; }
+  body.cid-training .col-container .col-nav:last-child{width: 100%;}
```
I would target the button here, specifically `a.button` to prevent overriding

Gauravpadam commented on static/css/training.css:
```diff
@@ -160,7 +180,8 @@ body.cid-training #get-certified .col-nav a.button {
     width: 100%;
   }
-  body.cid-training .col-container { flex-direction: column; }
+  body.cid-training .col-container {display: flex; flex-direction: column; }
+  body.cid-training .col-container .col-nav:last-child{width: 100%;}
```
same feedback is applicable here

Gauravpadam reviewed (commented):
Thanks

Here's some more feedback

Community-Programmer commented:
**Changes Made:**

- Changed the logo placement of cta.
- Added responsiveness.

![image](https://github.com/kubernetes/website/assets/76515568/24187e00-acbe-406a-b0c0-00f8566ebfb9)

aj11anuj reviewed (approved):
New icon alignment with responsiveness of icons and tiles, LGTM 👍

Squash all your commits into 2 commits:-
- All "CSS change" related commits into 1
- All "Addition of KCSA" related commits into 1

sftim commented:
/hold

Unusually, we'd like this squashed to 2 commits. The existing commit message “_Solved kubernetes#43317 KCSA Certification to Training page_” should **not** appear in the final commit list.

OK to unhold once that's sorted.

If you want @Community-Programmer, a maintainer can do that squash for you and update your fork accordingly. Let us know if you'd like that.

Community-Programmer commented:
> /hold
> 
> Unusually, we'd like this squashed to 2 commits. The existing commit message “_Solved #43317 KCSA Certification to Training page_” should **not** appear in the final commit list.
> 
> OK to unhold once that's sorted.
> 
> If you want @Community-Programmer, a maintainer can do that squash for you and update your fork accordingly. Let us know if you'd like that.

@sftim Yes, I would greatly appreciate it, please add a maintainer to do the squash and update my fork accordingly.Thank you

sftim commented:
Thanks, I squashed the commits and force pushed.

/lgtm
/hold cancel

Community-Programmer commented:
> [APPROVALNOTIFIER] This PR is **NOT APPROVED**
> 
> This pull-request has been approved by: _[aj11anuj](https://github.com/kubernetes/website/pull/43318#pullrequestreview-1660461883)_ **Once this PR has been reviewed and has the lgtm label**, please assign [reylejano](https://github.com/reylejano) for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).
> 
> The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=kubernetes%2Fwebsite).
> 
> Needs approval from an approver in each of these files:
> * **[OWNERS](https://github.com/kubernetes/website/blob/main/OWNERS)**
> 
> Approvers can indicate their approval by writing `/approve` in a comment Approvers can cancel approval by writing `/approve cancel` in a comment

/assign @reylejano

Community-Programmer commented:
@sftim can we merge it now??

Gauravpadam commented:
@Community-Programmer I would stay patient and wait for the approvers to get to this PR, Please refrain from tagging and/or assigning them. Thanks again!

aj11anuj commented:
It's almost ready to merge just a perspective of some other approver is awaiting.

reylejano commented:
/approve

tengqm commented:
/lgtm

sftim commented:
>  I would stay patient and wait for the approvers to get to this PR, Please refrain from tagging and/or assigning them. 

The assignment in https://github.com/kubernetes/website/pull/43318#issuecomment-1751298948 was **valid**; this was following advice from @k8s-ci-robot. It's always OK to follow the robot's advice (even if the advice is actually wrong, that means there was a bug in the robot and not a mistake by the person who relied on its advice).
